src/midiEditorCommands.cpp

```diff
@@ -994,6 +994,53 @@ void cmdMidiFilterWindow(Command *command) {
 	}
 }
 
+void maybePreviewCurrentNoteInEventList(HWND hwnd) {
+	auto focused = ListView_GetNextItem(hwnd, -1, LVNI_FOCUSED);
+	char text[50] = "\0";
+	// Get the text from the length column (2).
+	// If this column is empty, we aren't dealing with a note.
+	ListView_GetItemText(hwnd, focused, 2, text, sizeof(text));
+	if (strcmp(text, "\0") == 0) {
+		return;
+	}
+	MidiNote note;
+	// Convert length to project time. text is always in measures.beats.
+	auto length = parse_timestr_len(text, 0, 2);
+	text[0] = '\0';
+	// Get the text from the position column (1).
+	ListView_GetItemText(hwnd, focused, 1, text, sizeof(text));
+	// Convert this to project time. text is always in measures.beats.
+	note.start = parse_timestr_pos(text, 2);
+	note.end = note.start + length;
+	text[0] = '\0';
+	// Get the text from the channel column (3).
+	ListView_GetItemText(hwnd, focused, 3, text, sizeof(text));
+	note.channel = atoi(text);
+	text[0] = '\0';
+	// Get the text from the parameter (note name) column (5).
+	ListView_GetItemText(hwnd, focused, 5, text, sizeof(text));
+	string noteNameWithOctave { text };
+	static const string noteNames[] = {"C", "C#", "D", "D#", "E", "F",
+		"F#", "G", "G#", "A", "A#", "B"};
+	// Loop through noteNames in reverse order so the sharp notes are handled first.
+	for (int i = static_cast<int>(size(noteNames)) -1; i >= 0; --i) {
+		auto noteNameLength = noteNames[i].length();
+		if (noteNameWithOctave.compare(0, noteNameLength, noteNames[i]) != 0) {
+			continue;
+		}
+		int octave = stoi(noteNameWithOctave.substr(noteNameLength, string::npos));
+		note.pitch = ((octave + 1) * 12) + i;
+		break;
+	}
```

Comment:
What happens in the case where we don't match a note name (e.g. custom note names)? It looks like we preview the notes regardless. That seems potentially problematic.

Reply:
That's correct, as I didn't deal with that yet. I was searching for an opportunity to test these custom note names.